File: epi_judge_python/parity.py
from test_framework import generic_test


def parity(x: int) -> int:
    bit = 0
    while x != 0:
        x = x & (x - 1)
        bit ^= 1
    return bit


if __name__ == '__main__':
    exit(generic_test.generic_test_main('parity.py', 'parity.tsv', parity))

File: epi_judge_java/epi/Parity.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

public class Parity {
  @EpiTest(testDataFile = "parity.tsv")
  public static short parity(long x) {
    short parity = 0;

    while (x != 0) {
      x &= x - 1;
      parity ^= 1;
    }

    return parity;
  }

  public static void main(String[] args) {
    System.exit(
        GenericTest
            .runFromAnnotations(args, "Parity.java",
                new Object() {
                }.getClass().getEnclosingClass())
            .ordinal());
  }
}
